foundry/foundry.py

```diff
@@ -100,19 +114,35 @@ def load(self, name, download=True, globus=True, verbose=False, **kwargs):
         -------
             self
         """
+        res = []
         # MDF specific logic
-        res = self.forge_client.match_field(
-            "mdf.organizations", "foundry"
-        ).match_resource_types("dataset")
-        res = res.match_field("mdf.source_id", name).search()
+        # Handle DOI inputs
+        if name.startswith("10.") and provider == "MDF":
+            res = (
+                self.forge_client.match_dois(name)
+                .match_resource_types("dataset")
+                .match_field("mdf.organizations", "foundry")
+                .search()
+            )
 
-        # TODO: if object empty, handle
-        res = res[0]
-        res["dataset"] = res["projects"]["foundry"]
-        res["dataset"]["type"] = res["dataset"]["package_type"]
-        del res["projects"]["foundry"]
+        # Handle MDF source_ids
+        else:
+            print("source_id")
+            res = (
+                self.forge_client.match_field("mdf.organizations", "foundry")
+                .match_resource_types("dataset")
+                .match_field("mdf.source_id", name)
+                .search()
+            )
 
-        self = Foundry(**res)
+        if not res:
+            return self
+        else:
+            res = res[0]
+            res["dataset"] = res["projects"]["foundry"]
+            res["dataset"]["type"] = res["dataset"]["package_type"]
+            del res["projects"]["foundry"]
+            self = Foundry(**res)
```

Comment:
also unclear to me what's going on here, we replace the existing instance??

Reply:
Yes, I think this was because when I was trying to re-set values within the self object they weren't sticking. This is probably not the best way to do it...

Reply:
gotcha, my concern here is that this may result in some unexpected behavior and tricky bugs down the road. I might just wanna flag it for potential re-working in the future.

Reply:
I'll add comments/TODOs